substratest/assets.py

```diff
@@ -60,7 +104,11 @@ def future(self):
         """Returns future from asset."""
         assert hasattr(self, 'status')
         assert hasattr(self, 'key')
-        return Future(self, self._session)
+
+        try:
+            return self.Meta.FutureCls(self, self._session)
+        except AttributeError:
+            return Future(self, self._session)
```

Comment:
I think it will be simpler to create 2 classes: `_FutureMixin`, `_ComputePlanFutureMixin` instead of using meta attributes. For instance:
```python


class _FutureMixin(abc.ABC):
    _future_class = Future

    def attach(self, session):
        """Attach session to asset."""
        self._session = session
        return self

    def future(self):
        """Returns future from asset."""
        assert hasattr(self, 'status')
        assert hasattr(self, 'key')
        return self._future_class(self, self._session)

class _ComputePlanFutureMixin(_FutureMixin):
    _future_class = Future
```

Reply:
Ok!